internal/web_server/web/handlers/quiz_pages_handlers.go

```diff
@@ -45,14 +46,28 @@ func (qph *QuizPagesHandler) quizHomePage(c echo.Context) error {
 	if err != nil {
 		return err
 	}
+	userRankingInfo := user_ranking.UserRanking{}
 
-	userRankingInfo, err := user_ranking.GetUserRanking(qph.sharedData.DB, utils.GetUserIDFromCtx(c))
+	userRankingInfo, err = user_ranking.GetUserRanking(qph.sharedData.DB, utils.GetUserIDFromCtx(c))
 	if err != nil {
-		return err
+		if err == sql.ErrNoRows {
+			user, err := users.GetUserByID(qph.sharedData.DB, utils.GetUserIDFromCtx(c))
+			if err != nil {
+				return err
+			}
+			userRankingInfo = user_ranking.UserRanking{
+				Username:  user.Username,
+				Points:    0,
+				Placement: 0,
```

Comment:
technically this can be moved into the GetUserRanking function, and it can return the username with 0 points and placement 0 with no error. would make the handler thinner 
not a big deal tho